tests/unit/test_endpoint_consumer.py

```diff
@@ -138,6 +136,11 @@ class TestEndpointConsumer(unittest.TestCase):
     def setUp(self):
         metadata_file = open("metadata.yaml")
         self.harness = Harness(EndpointConsumerCharm, meta=metadata_file)
+
+        self.harness.add_resource(
+            "promql-transform-amd64",
```

Comment:
Wouldn't this break in the (possibly unlikely) event someone executes these tests on "arm" even though you are patching `platform.processor()` ? Though we can ignore this for beta of course.

Reply:
That is correct. Right now the tests require amd64, which I'm totally fine with for now.